lib/internal/Magento/Framework/Url.php

```diff
@@ -488,6 +485,26 @@ public function getBaseUrl($params = [])
         return $result;
     }
 
+    /**
+     * Set Requested scope, or custom scope if it sets on params.
+     *
+     * @param array|null $params
+     * @return \Magento\Framework\Url\ScopeInterface
+     */
+    private function setRequestScope($params = null)
+    {
+        $customScope = $this->getRouteParamsResolver()->getRouteParam('_scope');
+
+        if (isset($params['_scope'])) {
```

Comment:
What do you think about introducing a `setScope` method to `\Magento\Backend\Model\Url` to be able to assign custom scope to `_scope` property? (it's better to also keep parent method call for backward compatibility purposes)

Reply:
@sivaschenko yes, that's what i needed, now it looks fine, i'm little bit confused by overriding method _getScope()